pkg/reconciler/v1alpha1/testing/functional.go

```diff
@@ -122,6 +133,45 @@ func WithServiceLabel(key, value string) ServiceOption {
 	}
 }
 
+// WithResourceRequirements attaches resource requirements to the service
+func WithResourceRequirements(resourceRequirements corev1.ResourceRequirements) ServiceOption {
+	return func(svc *v1alpha1.Service) {
+		svc.Spec.RunLatest.Configuration.RevisionTemplate.Spec.Container.Resources = resourceRequirements
+	}
+}
+
+// AddVolume adds a volume to the service
+func AddVolum(name, mountPath string, volumeSource corev1.VolumeSource) ServiceOption {
+	return func(svc *v1alpha1.Service) {
+		rt := &svc.Spec.RunLatest.Configuration.RevisionTemplate.Spec
+
+		rt.Container.VolumeMounts = []corev1.VolumeMount{{
+			Name:      name,
+			MountPath: mountPath,
+		}}
+
+		rt.Volumes = []corev1.Volume{{
+			Name:         name,
+			VolumeSource: volumeSource,
+		}}
+	}
+}
+
+// WithoutAutoScaling turns off auto scaling by setting max and min scale to 1
+func WithoutAutoScaling(service *v1alpha1.Service) {
```

Comment:
This seems to be used only once, do we need to pull it out into an extra function?

We could add `WithConfigAnnotation` to make it easier to set these values. WDYT?

Reply:
Making the func more generalized and more likely to be reused certainly wouldn't hurt.